aio_pika/queue.py

```diff
@@ -63,7 +66,10 @@ def declare(self, timeout: int = None, passive: bool = False) -> asyncio.Future:
         )
 
         def on_queue_declared(result):
-            self.name = result.result().method.queue
+            method = result.result().method
+            self.name = method.queue
+            self.message_count = method.message_count
+            self.consumer_count = method.consumer_count
```

Comment:
An example alternative would be to add a `__slot__` called, e.g., `declare_result` and store the entire `queue-ok` object in there.